dev/ProgressBar/ProgressBar.cpp

```diff
@@ -93,32 +94,47 @@ void ProgressBar::UpdateStates()
     }
     else if (!IsIndeterminate())
     {
-        SetProgressBarIndicatorWidth();
         winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
     }
 }
 
 void ProgressBar::SetProgressBarIndicatorWidth()
 {
+    const auto templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+
     if (auto&& progressBar = m_layoutRoot.get())
     {
         if (auto&& progressBarIndicator = m_progressBarIndicator.get())
         {
             const double progressBarWidth = progressBar.ActualWidth();
+            const double prevIndicatorWidth = progressBarIndicator.ActualWidth();
             const double maximum = Maximum();
             const double minimum = Minimum();
             const auto padding = Padding();
 
-            if (std::abs(maximum - minimum) > DBL_EPSILON)
+            m_isUpdating = true; // Adds "Updating" state in between to trigger RepositionThemeAnimation Visual Transition in ProgressBar.xaml when reverting back to previous state
```

Comment:
Minor issue: Use a RAII helper to ensure that m_isUpdating gets set back to false, even in the case where an exception is raised by the below code.

e.g.
m_isUpdating = true; 
auto scopeGuard = gsl::finally([this]()
{
    m_isUpdating = false; 
});

Reply:
If you go with the direct call to GoToState, this will be unnecessary, since there would no longer be the m_isUpdating member variable.